### PR TITLE
Enhanced Semaphores

### PR DESCRIPTION
**Pull request by jnealtowns**
Reviewer: @rlane 

- Timeout support
- Proper error condition handling.

### CODE REVIEW

File: modules/OS/module/src/os_sem_posix.c [hunk 1]
```diff
@@ -40,26 +42,89 @@ os_sem_create(int count)
...
+}
+
+
+/**
+ * Initialize a timespec structure for the given timeout interval.
+ */
+static void
+timespec_init_timeout__(struct timespec* ts, uint64_t us)
+{
+    clock_gettime(CLOCK_REALTIME, ts);
```
Comment (rlane) [inline modules/OS/module/src/os_sem_posix.c]: `sem_timedwait` uses gettimeofday which I assume is the same as CLOCK_REALTIME.

**Comment by rlane** [comment]
ACK

**Comment by bsn-abat** [comment]
ABAT: START_MERGE:  Log file at http://vbs22/abat/2015.01.22.2310-m.bigcode.master/abat.log

**Comment by bsn-abat** [comment]
ABAT: AUTO-REJECT:  Your change failed pre-merge validation defined in bigcode/.abat-automerge. See log for details.


INSTRUCTIONS: Please fix your problem by

1) Reviewing the logs below and `git commit`ing a fix.

2) Use `git push` to add new commits to this same remote branch/pull request.

3) Get another code review approval from your co-worker indicated by a line of "ACK" in comment, or add another comment yourself with a line of "R_E_T_R_Y (Remove the underscores)". That will cause abat to try to merge your code again.



Log file at http://vbs22/abat/2015.01.22.2310-m.bigcode.master/abat.log
/cc @rlane @jnealtowns

**Comment by jnealtowns** [comment]
RETRY

**Comment by bsn-abat** [comment]
ABAT: START_MERGE:  Log file at http://vbs22/abat/2015.01.23.0012-m.bigcode.master/abat.log

**Comment by bsn-abat** [comment]
ABAT: ACCEPT:  Successfully merged

In case you want to see the build log, check out :
Log file at http://vbs22/abat/2015.01.23.0012-m.bigcode.master/abat.log
/cc @rlane @jnealtowns

**Pull request merged**